src/main/java/com/powsybl/openloadflow/ac/TransformerVoltageControlOuterLoop.java

```diff
@@ -14,13 +14,18 @@
 import com.powsybl.openloadflow.network.LfNetwork;
 import com.powsybl.openloadflow.network.TransformerVoltageControl;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * @author Anne Tilloy <anne.tilloy at rte-france.com>
  */
 public class TransformerVoltageControlOuterLoop extends AbstractTransformerVoltageControlOuterLoop {
 
     private double maxControlledNominalVoltage = Double.MIN_VALUE;
 
+    private List<LfBus> disabledControllerBuses = new ArrayList<>();
```

Comment:
Disabled but never restored?
Also, we have to think about state management in outer loops, we should keep them stateless if possible to be able later to parallelize the security analysis more easily.